include/gatekeeper_lls.h

```diff
@@ -19,6 +19,183 @@
 #ifndef _GATEKEEPER_LLS_H_
 #define _GATEKEEPER_LLS_H_
 
-int run_lls(void);
+#include <rte_timer.h>
+
+#include "gatekeeper_mailbox.h"
+#include "gatekeeper_net.h"
+
+/*
+ * Maximum key length (in bytes) for an LLS map. It should be set
+ * to the maximum key length needed. Currently, this is set to 16 for
+ * the number of bytes in an IPv6 address.
+ */
+#define LLS_MAX_KEY_LEN (16)
+
+/* Number of records that a LLS cache can hold. */
+#define LLS_CACHE_RECORDS (1024)
+
+/* Requests that can be made to a cache. */
+enum lls_req_ty {
+	/* Do a one-time resolution; do not receive further updates. */
+	LLS_REQ_RESOLVE,
```

Comment:
So if `call_again` is false, then the hold can be removed? In other words, it's an implicit put?

Then, requests of `LLS_REQ_PUT` will only need to remove a hold if one exists.